lib/msf/core/exe/segment_injector.rb

```diff
@@ -15,8 +15,20 @@ def initialize(opts = {})
       @payload = opts[:payload]
       @template = opts[:template]
       @arch  = opts[:arch] || :x86
-      @buffer_register = opts[:buffer_register] || 'edx'
-      unless %w{eax ecx edx ebx edi esi}.include?(@buffer_register.downcase)
+      @buffer_register = opts[:buffer_register]
+
+      x86_regs = %w{eax ecx edx ebx edi esi}
+      x64_regs = %w{rax rcx rdx rbx rdi rsi} + (8..15).map{|n| "r#{n}" }
+
+      @buffer_register ||= if @arch == :x86
```

Comment:
Why not terniary assignment? 
``
@buffer_register = (@arch == :x86) ? "edx" : "rdx"
``